src/store/page-load/saga.test.ts

```diff
@@ -0,0 +1,106 @@
+import { expectSaga } from 'redux-saga-test-plan';
+import { saga } from './saga';
+import { getHistory } from '../../lib/browser';
+import { rootReducer } from '../reducer';
+import { getCurrentUserWithChatAccessToken } from '../authentication/saga';
+import { call } from 'redux-saga/effects';
+import { initializePublicLayout } from '../layout/saga';
+
+jest.mock('../../config', () => ({
+  config: {
+    defaultZnsRoute: 'wilder',
+    defaultApp: 'nfts',
+  },
+}));
+
+jest.mock('../../lib/feature-flags', () => ({
+  featureFlags: {
+    allowPublicZOS: false,
+  },
+}));
+
+class StubHistory {
+  pathname: string;
+  location: object;
+
+  constructor(pathname?: string) {
+    this.pathname = pathname ?? '';
+    this.location = { pathname };
+  }
+
+  replace = jest.fn();
+}
+
+describe('page-load saga', () => {
+  it('redirects to main page if user is present tries to access login/signup page', async () => {
+    const initialState = { pageload: { isComplete: false } };
+
+    // login
+    let history = new StubHistory('/login');
+    const { storeState: loginStoreState } = await expectSaga(saga)
+      .withReducer(rootReducer, initialState as any)
+      .provide(stubResponses(history, true))
+      .run();
+
+    // redirected from /login to /0.wilder.nfts
+    expect(history.replace).toHaveBeenCalledWith({ pathname: '/0.wilder/nfts' });
+    expect(loginStoreState.pageload.isComplete).toBe(false);
```

Comment:
Interesting, I would have expected this to be `true` at the end if everything was successful. We're done with verifying the initial page load no matter which final condition is reached, right?

Reply:
Now that i think about it, it should be `true`. Previously we were just setting the value & returning. I have updated the code and the test.